### DISCUSSION TITLE
Great Ideas & Roadmaps

### DISCUSSION THREAD

**Discussion by lengockhanhnk** [SakanaAI/AI-Scientist#12] [General]
Could you show the roadmaps to develop it in the future?

**Comment by juhonkang** [comment]
Great

**Comment by conglu1997** [comment]
Please see the future work section of our paper!

**Reply by BradKML** [reply]
Maybe instead of only using the roadmap, the public can also throw in their own ideas on how things could improve?

**Reply by conglu1997** [reply]
Indeed, we welcome anything :)

**Comment by BradKML** [comment]
Here are some small ideas regarding the making and testing of new models:
1. It needs to anticipate Mamba-esque LLM, Diffusion-based Language Models, and alternate image generation architectures (together multi-modal architectures would have variety) https://github.com/bansky-cl/diffusion-nlp-paper-arxiv https://github.com/XiudingCai/Awesome-Mamba-Collection
2. Hyperparameter optimization (like the Ray library or the papers on predicting ideal hyperparameters) and things like Grokfast should be included to speed up finding the best solutions https://github.com/ironjr/grokfast
3. Pruning and merging should be considered with their own toolkit (not just previous blog posts but also research similar to ones by PrunaAI) https://github.com/liyunqianggyn/Awesome-LLMs-Pruning

Maybe some more on: dataset gathering and pipelines, or prompt fuzz-testing (formatting optimization or adversarial attacks), or diverse benchmarking and subjective goal preferences (MOO).

**Comment by bdutta19** [comment]
we are trying to build a stand alone UI over Ai-scientist.. right now it is simple Gradio app but once our ux is settled we plan to make it more robust ..
![image](https://github.com/user-attachments/assets/40129f56-08b2-4aa6-bcc1-d3cfc8a1b983)

**Reply by BradKML** [reply]
Any effort in that direction is appreciated, since having something akin to A1111 is easier than having something closer to ComfyUI with all the drag-and-drop (e.g. Dify). Can it generate and test experiments tho?

**Comment by BradKML** [comment]
Found some alternative code that might be good to learn from https://github.com/Technion-Kishony-lab/data-to-paper
